src/main/scala/ai/datahunters/md/workflow/MainExtractionWorkflow.scala

```diff
@@ -34,13 +35,23 @@ class MainExtractionWorkflow(config: ProcessingConfig,
     val columnNamesConverter = ColumnNamesConverterFactory.create(config.namingConvention)
     val rawInputDF = reader.load()
     val hashExtractor = config.hashList.map(s => HashExtractor(s))
-    val pipeline = ProcessingPipeline(rawInputDF)
+
+    val hashGenerationPipeline = ProcessingPipeline(rawInputDF)
+    hashExtractor.foreach(hashGenerationPipeline.addProcessor)
+    val dfWithHashes = hashGenerationPipeline.run()
+
+    val solrHashDF = solrHashReader.map(r => r.load())
+    val hashComparator = HashComparator(solrHashDF, config.hashList)
```

Comment:
@dawidsau I thinkg we should add this processor only if this feature was enabled by user. What do you think?